### PR TITLE
AN-5013/eth-sl-2 (do not merge)

### PR DESCRIPTION
**Pull request by drethereum**
Ethereum Strealime 2.0 Upgrade
- Includes use of `fsc_evm` macro package

1. Update API_INTEGRATION in dbt_project.yml (and remove temp realtime model if necessary)
2. Merge [snowflake-streamline ](https://github.com/FlipsideCrypto/streamline-snowflake/pull/159)+ [fsc-utils](https://github.com/FlipsideCrypto/fsc-utils/pull/49), then this PR
3. Pull in main + run `dbt clean && dbt deps` (if running following commands locally)
4. `DROP SCHEMA ethereum._live;` + `dbt run -m livequery_models.deploy.core --vars '{UPDATE_UDFS_AND_SPS: true}' --target prod`
5. Grant usage on schema/all functions to `DBT_CLOUD_ETHEREUM` and `AWS_LAMBDA_ETHEREUM_API` roles + `GRANT USAGE ON INTEGRATION AWS_ETHEREUM_API_PROD_V2 TO ROLE DBT_CLOUD_ETHEREUM;`
6. Run `dbt run-operation fsc_utils.create_evm_streamline_udfs --vars '{UPDATE_UDFS_AND_SPS: true}' --target prod`
7. Run `dbt run -m models/streamline/bronze models/streamline/silver/abis/complete models/streamline/silver/balances/complete models/streamline/silver/beacon/complete models/streamline/silver/core/complete models/streamline/silver/decoder/complete models/streamline/silver/reads/complete models/streamline/silver/core/streamline__get_chainhead.sql --full-refresh`

### CODE REVIEW

File: dbt_project.yml [hunk 1]
```diff
@@ -70,7 +70,32 @@ vars:
   HEAL_MODELS: []
   LOAD_CUSTOM_FUNCTIONS: False
   START_GHA_TASKS: False
-  API_INTEGRATION: '{{ var("config")[target.name]["API_INTEGRATION"] if var("config")[target.name] else var("config")["dev"]["API_INTEGRATION"] }}' 
+
+#### STREAMLINE 2.0 BEGIN ####
+
+  API_INTEGRATION: '{{ var("config")[target.name]["API_INTEGRATION"] if var("config")[target.name] else var("config")["dev"]["API_INTEGRATION"] }}'
   EXTERNAL_FUNCTION_URI: '{{ var("config")[target.name]["EXTERNAL_FUNCTION_URI"] if var("config")[target.name] else var("config")["dev"]["EXTERNAL_FUNCTION_URI"] }}'
+  API_AWS_ROLE_ARN: '{{ var("config")[target.name]["API_AWS_ROLE_ARN"] if var("config")[target.name] else var("config")["dev"]["API_AWS_ROLE_ARN"] }}'
```
Comment (austinFlipside) [inline dbt_project.yml:78]: Julius wants the api integration to be his teams responsibility. We should remove the ARN and any code for creating API integrations. We do want to continue using them though in creating the external functions, to be clear

File: dbt_project.yml [hunk 2]
```diff
@@ -70,7 +70,32 @@ vars:
...
   ROLES: |
-      ["INTERNAL_DEV"]
+    ["INTERNAL_DEV"]
+
+  config:
+  # The keys correspond to dbt profiles and are case sensitive
+    dev:
+      API_INTEGRATION: AWS_ETHEREUM_API_STG_V2
+      EXTERNAL_FUNCTION_URI: p6h6x0ncl1.execute-api.us-east-1.amazonaws.com/stg/
+      API_AWS_ROLE_ARN: arn:aws:lambda:us-east-1:704693948482:function:ethereum-api-stg-APIHandler-EreAJfUDzpA180
```
Comment (austinFlipside) [inline dbt_project.yml:87]: same

File: models/silver/core/silver__blocks.sql [hunk 3]
```diff
@@ -48,7 +51,7 @@ SELECT
     ) :: INT AS total_difficulty,
     ARRAY_SIZE(
         DATA :result :transactions
-    ) AS tx_count,
+    ) AS tx_count, --update
```
Comment (austinFlipside) [inline models/silver/core/silver__blocks.sql:54]: what is this update?

File: dbt_project.yml (hunk 1 above)
Comment (drethereum) [reply dbt_project.yml:78]: how do we access / reference it if it's not in dbt_project.yml?

File: models/silver/core/silver__blocks.sql (hunk 3 above)
Comment (drethereum) [reply models/silver/core/silver__blocks.sql:54]: originally thought an update was needed here but I confirmed the array_size function still works as expected with the new blocks_transactions call structure, so I just need to remove the comment

File: dbt_project.yml (hunk 1 above)
Comment (austinFlipside) [reply dbt_project.yml:78]: Where do we access it? The api_integration macro isn't needed anymore per Julius

Comment (drethereum) [reply dbt_project.yml:78]: ah youre right we only reference the external function uri now - is the `create_aws_api_integrations()` macro in `fsc-utils` being used anywhere anymore then? if not we should remove that too

Comment (austinFlipside) [reply dbt_project.yml:78]: Good call, lets drop it in `fsc_utils` too. It isn't used in the larger evm macros but good hygiene to remove

**Review by austinFlipside — APPROVED** [review]

**Pull request merged**